### PR TITLE
Speed-up test execution on jenkins.

### PR DESCRIPTION
**Pull request by bohdan-suprun**
## Purpose
Speed-up test execution on jenkins by running tests in parallel (via JVM fork) and decreasing amount of log outputs.

image

## Approach
1. Run tests in parallel using `maven-surefire-plugin` fork capabilities (running separate JVM processes).
1. Add minor changes, so that `FakeOkapi` and `CirculationVerticle` uses dynamic port.
1. Use `debug` logging level in `ResponseInterpreter.java`, so each received response is not logged.
1. Configure `RestAssuredClient` to log only failed requests and responses by replacing `log(),all()` with `log().ifValidationFails()` and `log().greaterThan(300)`.

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

**Comment by bohdan-suprun** [comment]
Can not cover the line, because `debug` is disabled.

File: src/test/java/api/support/RestAssuredClient.java [hunk 1]
```diff
@@ -43,7 +44,7 @@ public static RestAssuredClient defaultRestAssuredClient() {
 
   public RequestSpecification beginRequest(String requestId) {
     return given()
-      .log().all()
+      .log().ifValidationFails()
```
Comment (marcjohnson-kint) [inline src/test/java/api/support/RestAssuredClient.java:47]: What validation does this refer to?

**Review by marcjohnson-kint — APPROVED** [review]

File: src/test/java/api/support/RestAssuredClient.java (hunk 1 above)
Comment (bohdan-suprun) [reply src/test/java/api/support/RestAssuredClient.java:47]: Status code of response I think.

Comment (marcjohnson-kint) [reply src/test/java/api/support/RestAssuredClient.java:47]: > Status code of response I think.

As I understand it, only some of the RestAssured request set expectations about what an acceptable status code is. What happens for the requests that do not specify that?

Comment (bohdan-suprun) [reply src/test/java/api/support/RestAssuredClient.java:47]: For this case I use `log().ifStatusCodeMatches(greaterThan(300))` - response is logged when status code > 300

**Pull request merged**